src/main/java/net/snowflake/ingest/streaming/internal/ChannelCache.java

```diff
@@ -65,7 +65,7 @@ void removeChannelIfSequencersMatch(SnowflakeStreamingIngestChannelInternal chan
           // We need to compare the channel sequencer in case the old channel was already been
           // removed
           return channelInCache != null
-                  && channelInCache.getChannelSequencer() == channel.getChannelSequencer()
+                  && channelInCache.getChannelSequencer().equals(channel.getChannelSequencer())
```

Comment:
Why this change out of curiosity? These should be `long`

Reply:
Yes and the reason is that `==` on objects checks if the object is the same while `equals()` (if correctly implemented) checks if the value is the same, irrespective of the object that caries it. So this could lead to bugs.